WooCommerce/WooCommerceUITests/Tests/LoginTests.swift

```diff
@@ -30,4 +30,22 @@ class LoginTests: XCTestCase {
 
         XCTAssert(WelcomeScreen.isLoaded())
     }
+
+    func testLoginOptionsArePresent() {
+        //verify tapping "Log in" will show options to enter a WordPress.com email, log in with Google, or enter your site address.
+        WelcomeScreen()
+        .selectLogin()
+        .openHelpMenu()
+        //will add .sendEmailToSupport once I have some questions answered about that
+        .closeHelpMenu()
+
+        //verify all login options exist
+        .emailLoginOption()
+        .siteAddressLoginOption()
+        .googleLoginOption()
```

Comment:
1. gotcha
2. Am I on the right track to put them all in a `.isLoaded` method for the LoginEmailScreen? Or is there an even better solution?

Reply:
> Am I on the right track to put them all in a .isLoaded method for the LoginEmailScreen? Or is there an even better solution?

You're on the right track to think about it as a question of whether the screen is fully loaded with all expected elements. However, you can make the necessary assertions when the page object is initialized — that way whenever you expect that screen to be returned in a test, it'll ensure the screen includes all those elements. You can see an example of that in the `init()` method on the [PeriodStatsScreen](https://github.com/woocommerce/woocommerce-ios/blob/5627c70af2219f62e89ebd936f6a7e460cc154c4/WooCommerce/WooCommerceUITests/Screens/MyStore/PeriodStatsTable.swift#L23), which checks to make sure all the expected stats tabs are loaded (on the logged-in My Store screen).

However, another way to think about it: Once we have tests that go through each of the user flows, we could drop the explicit assertions and know that if the option isn't available on that screen the corresponding test will fail. That will make the tests more efficient but still maintain those checks somewhere in the process. (Since we don't have those tests automated yet, it is fair to check for those options explicitly e.g. in the `init()` method.)

Reply:
I believe I corrected this correctly! :D   Thanks for such a great explanation!